app/experimenter/experiments/constants/nimbus.py

```diff
@@ -128,6 +129,15 @@ class PublishStatus(models.TextChoices):
     }
     PUBLISH_STATUS_ALLOWS_UPDATE = (PublishStatus.IDLE,)
 
+    # State Predicates
+    IS_LAUNCHING = Q(status=Status.DRAFT, publish_status=PublishStatus.WAITING)
+    IS_ENDING = Q(
+        status=Status.LIVE,
+        publish_status=PublishStatus.WAITING,
+        is_end_requested=True,
+    )
```

Comment:
FWIW, these might actually be handy as model methods if we plumb them through GQL to the frontend as booleans. Then the frontend won't need to duplicate this logic. Not a huge deal either way though

Reply:
Yeah that's a good point this might not scale to the frontend, we can keep an eye out for that.